### ISSUE TITLE
Automate release process 

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#3365]
generateALL will do generation of all posts

1. On release time I review chaages in xdoc post and make a decision about next version number
2. if version need to changed I do "mvn:version set" and let Travis regenerate xdoc
3. if nochanges required, I copy xdoc post as it is and make a commit.
4. release trigger is "mvn release:prepare" commits
5. Travis should trigger "mvn release:perform" and do publication to {mvn repo, site, sorceforge downloads}={release.sh} and do publications to social (build need to be executed on source strict version, not a SNAPSHOT)

How to triger Travis on strict version ?
on commit - parse pom.xml and catch non SNAPSHOT verson.
from release.sh:
if (xmlstarlet sel -N pom=http://maven.apache.org/POM/4.0.0 -t -m pom:project -v pom:version pom.xml | grep SNPASHOT | cat ) != ""

we can already use this command to prevent Travis failures during relase!!!

PROBLEMs !!!!!
1) Problem is that all logins/password and gpg keys/passphrases are placed in settings.xml but should be taken from environment variables.

2) Right now copy to Sourceforge for web site and binaries is done by scp, to script it will require 
to exchange by keys , that might be a problem as Sourceforge does not provide permanent ssh access, it should be requested by "ssh -t create ...".

can maven resolve environment variables in settings.xml ?
can we provide all that in commandline ?

we could potentially run release on sourseforge servers (we have permanent disk storage there).

**Comment by MEZk** [comment]
@romani 
>can maven resolve environment variables in settings.xml ?

Yes, it can. See "Properties" section https://maven.apache.org/settings.html

> ... values are accessible anywhere within a POM by using the notation ${X}, where X is the property.

```xml


    <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                          https://maven.apache.org/xsd/settings-1.0.0.xsd">
      ...
      <profiles>
        <profile>
          ...
          <properties>
            <user.install>${user.home}/our-project</user.install>
          </properties>
          ...
        </profile>
      </profiles>
      ...
    </settings>
```
